Argo/Types/Decodable.swift

```diff
@@ -1,4 +1,4 @@
 public protocol Decodable {
   typealias DecodedType = Self
-  static func decode(json: JSON) -> Decoded<DecodedType>
+  static func decode(json: JSON) throws -> DecodedType
```

Comment:
~~`typealias DecodedType = Self`~~ and `static func decode(json: JSON) throws -> Self`?